kernel/kernel-api/src/main/java/io/delta/kernel/internal/stats/FileSizeHistogram.java

```diff
@@ -257,6 +260,30 @@ public Row toRow() {
     return new GenericRow(FULL_SCHEMA, value);
   }
 
+  public FileSizeHistogramResult captureFileSizeHistogramResult() {
```

Comment:
We want to expose as immutable of an API for accessing the final FileSizeHistogram which I've exposed as the interface `FileSizeHistogramResult`. We can't actually prevent anyone from mutating those arrays returned in the interface but we at the least make a copy here for best effort. (explanation as to why we have FileSizeHistogramResult -- we also need a public API to expose to the engine in the metrics result interface)